src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1007,23 +1075,41 @@ query_path(const char *szInput, int *is_target_path, dfs_obj_t **parent, char *i
 		/* trying to avoid lock as much as possible */
 		if (!daos_inited) {
 			/* daos_init() is expensive to call. We call it only when necessary. */
+
+			D_MUTEX_LOCK(&lock_global);
+
+			rc = consume_low_fd();
+			if (rc) {
+				DS_ERROR(rc, "consume_low_fd() failed");
+				*is_target_path = 0;
+				D_MUTEX_UNLOCK(&lock_global);
+				goto out_normal;
+			}
 			rc = daos_init();
 			if (rc) {
 				DL_ERROR(rc, "daos_init() failed");
 				*is_target_path = 0;
+				D_MUTEX_UNLOCK(&lock_global);
 				goto out_normal;
 			}
+
 			if (eq_count_max) {
-				rc = daos_eq_create(&td_eqh);
-				if (rc)
-					DL_WARN(rc, "daos_eq_create() failed");
-				main_eqh = td_eqh;
-				rc       = pthread_atfork(NULL, NULL, &child_hdlr);
+				D_MUTEX_LOCK(&lock_eqh);
```

Comment:
i don't understand why we need nested locks now

Reply:
ok. I will rearrange the code to avoid nested locks. Thank you!